controllers/machinedeployment_controller.go

```diff
@@ -193,7 +193,7 @@ func (r *MachineDeploymentReconciler) getMachineSetsForDeployment(d *clusterv1.M
 			continue
 		}
 
-		if !selector.Matches(labels.Set(ms.Labels)) {
+		if !selector.Matches(labels.Set(ms.Labels)) && !metav1.IsControlledBy(ms, d) {
```

Comment:
We already have the check for `!metav1.IsControlledBy(ms, d)` on line 211 - how is this different?

Reply:
I think this is what I was asking for in #1341. It's a double negative, but inverted it's "skip this machine set unless either the selector matches or it has a controller ref pointing to this machine deployment".

Reply:
it doesn't ignore machinesets that do not have same labels after updating machinedeployment labels

Reply:
Thanks, I see the subtle difference now. Perhaps some comments would be in order?

Reply:
I've added a comment